matrix/adding-rooms.md

```diff
@@ -0,0 +1,64 @@
+# Matrix Room Onboarding
+
+The `@admin:nixos.org` user is capable of adding an alias that is rooted on its own homeserver, which is what we want for branding reasons.
+
+It is however not a server admin, so we cannot rely on https://github.com/grahamc/matrix-management-tools/blob/main/process-all-rooms.sh, which requires access to the `/_synapse/` Admin API.
+
+We can deal with both new and pre-existing rooms. For pre-existing rooms the existing admins need to hand over that permissions, and demote to moderator, so the room will fall exclusively under community administration.
+
+## Preconditions
+
+- Be joined to `#moderators:nixos.org`
+    - so you can control Mjölnir
+- Active session on the `@admin:nixos.org` account
+    - The secret recovery key for the encryption things is on bitwarden
+- Get moderator permissions on the space (or subspace) you want to add the room to
+    - `!mjolnir powerlevel @admin:nixos.org 50 <room>`
+
+## Spaces
+
+Making a decision in which space or subspace a room belongs
+
+- Main space
+    - Community
+        - Community projects
+    - International
+        - Regional channels
+    - Teams
+        - Official Teams
+        - Unclear what they map to
+            - https://nixos.org/community/#governance-teams
+            - https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix
+            - https://github.com/orgs/NixOS/teams
+    - Topics
+        - Things that are not official teams yet? Best guess.
+
+## Tasks
+
+From the `@admin:nixos.org` user, execute the following steps:
+
+- Create new room
+    - or join it, if it already exists
+- If you don't have "Admin" permissions, request them from existing admins
+- Ask existing admins to step down to moderator role
+    - Rooms should be owned by the community, represented by the moderation team, so we can guarantee the room will be maintained going forward
+    - We can probably be more lenient for community project rooms
```

Comment:
Do we all agree on this?

Reply:
Still unclear to me what this means. Lenient in what aspect?

Reply:
Leave community projects with admin permissions to the room, as opposed to asking them to step down to moderator.

Imagine you'd want disko on the nixos space. We'd add Mjölnir as admin and ask you to step down to moderator. Yay/nay?

Reply:
The process I've been following in the meantime is to give international rooms the choice of whether they want to stay as admins, and to give community rooms the choice of whether they want to stay as admins, _and_ whether they want mjolnir at all.  I think this is a good balance of the proximity of those types of rooms to the project.

Reply:
I think most people want to be at least able to set the topic. Don't know what powerlevel this requires. But guess if they own the project they should also own the channel. If they go rogue than we would probably need to remove the channel as a whole from the NixOS space.

Reply:
We can also give moderators extra permissions as required.  For example, I've given a moderator the power to change permissions before (which doesn't let them promote anybody higher than themselves).